test/libcommonserver/log/testlog.cpp

```diff
@@ -51,4 +47,78 @@ void TestLog::testLog() {
 
     CPPUNIT_ASSERT(true);
 }
+
+void TestLog::testLogRollingAndExpiration(void) {
+    // Clear the log directory
+    IoError ioError = IoErrorSuccess;
+    IoHelper::DirectoryIterator dirIt(_logDir, false, ioError);
+
+    bool endOfDirectory = false;
+    DirectoryEntry entry;
+    while (dirIt.next(entry, endOfDirectory, ioError) && !endOfDirectory && ioError == IoErrorSuccess) {
+        if (entry.path().filename().string() == Log::instance()->getLogFilePath().filename().string()) {
+            continue;
+        }
+        IoHelper::deleteDirectory(entry.path(), ioError);
+    }
+
+    testLargeLogRolling();
+    testExpiredLogFiles();
+}
+
+void TestLog::testLargeLogRolling(void) {
+    CPPUNIT_ASSERT_EQUAL(1, countFilesInDirectory(_logDir));
+    // Generate a log larger than the max log file size.
+    std::string testLog = "Test info log/";
+    for (int i = 0; i < 1000; i++) {
+        testLog += "Test info log/";
+    }
+    uint64_t currentSize = 0;
+    std::cout << std::endl
```

Comment:
I am afraid that these debug print statements are going to clutter the unit tests output.

Reply:
![image](https://github.com/Infomaniak/desktop-kDrive/assets/82284536/e106f9b6-c340-440f-80e2-cf894927c7ef)
As the process can be long 30s<t<60s (especially on windows) I think it can be a good think to have a visualization of the advancement?

Reply:
We may discuss this as a desktop dev team topic. I am worried of the impact on the readability of the unit tests output. When errors happen in unit tests, we identify them more easily if the output is kept simple. Besides, a test that lasts up to one minute does not look like a unit test anymore.

Reply:
I understand your concern regarding readability. To mitigate this impact, I propose an alternative approach: [commit 09737d1](https://github.com/Infomaniak/desktop-kDrive/pull/159/commits/09737d15dc2d63b0df49cabac27ee750a5acb4ae).

![image](https://github.com/Infomaniak/desktop-kDrive/assets/82284536/3e130b2e-3a58-4690-8bda-203cf7fe592e)

This solution is less invasive and ensures that the output is completely removed at the end of the test, maintaining the original readability state.

![image](https://github.com/Infomaniak/desktop-kDrive/assets/82284536/4c5af995-0529-4d04-b180-ac8dc3ab6055)

Could this approach be a good comromise?

Reply:
The unit test itself runs instantaneously. However, the preparation phase is time-consuming because it requires "spamming" with `LOG_DEBUG` calls until the maximum logfile size is reached. This process is relatively quick on Linux and macOS, but it becomes significantly more time-consuming on Windows.

Reply:
Regarding your proposal for reducing the noise in the unit tests output: thumbs up! As long as the final output is kept short and readable, I won't complain.

Regarding, the issue with the unit test preparation: would it be possible to keep the iteration count as low as possible? Two long debug lines?

Reply:
After a quick test, it seems that after 12000 characters there isn't any significant improvement by increasing the size of the log output.

Reply:
(Currently each line is 14000 characters long)